botocore/handlers.py

```diff
@@ -943,6 +945,7 @@ def remove_subscribe_to_shard(class_attributes, **kwargs):
     ('before-parameter-build.route53', fix_route53_ids),
     ('before-parameter-build.glacier', inject_account_id),
     ('after-call.s3.ListObjects', decode_list_object),
+    ('after-call.s3.ListObjectsV2', decode_list_object),
```

Comment:
So unfortunately. We cannot purely reuse the `decode_list_objects`. There are different keys that need to be decoded according to the [docs](https://docs.aws.amazon.com/AmazonS3/latest/API/v2-RESTBucketGET.html). Specifically, we need to decode: `Delimiter`, `Prefix`, `ContinuationToken`, `Key`, and `StartAfter`. I would recommend trying to see if we can consolidate the logic and make a new `decode_list_objects_v2` handler that decodes it for the V2 response shape and use that instead.